test/e2e/apps/controller_revision.go

```diff
@@ -0,0 +1,266 @@
+			}
+		}
+		if daemonsets, err := f.ClientSet.AppsV1().DaemonSets(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{}); err == nil {
+			framework.Logf("daemonset: %s", runtime.EncodeOrDie(scheme.Codecs.LegacyCodec(scheme.Scheme.PrioritizedVersionsAllGroups()...), daemonsets))
+		} else {
+			framework.Logf("unable to dump daemonsets: %v", err)
+		}
+		if pods, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{}); err == nil {
+			framework.Logf("pods: %s", runtime.EncodeOrDie(scheme.Codecs.LegacyCodec(scheme.Scheme.PrioritizedVersionsAllGroups()...), pods))
+		} else {
+			framework.Logf("unable to dump pods: %v", err)
+		}
+		err = clearDaemonSetNodeLabels(f.ClientSet)
+		framework.ExpectNoError(err)
+	})
+
+	f = framework.NewDefaultFramework("controllerrevisions")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
+
+	image := WebserverImage
+	dsName := "e2e-" + utilrand.String(5) + "-daemon-set"
+
+	var ns string
+	var c clientset.Interface
+
+	ginkgo.BeforeEach(func() {
+		ns = f.Namespace.Name
+
+		c = f.ClientSet
+
+		updatedNS, err := patchNamespaceAnnotations(c, ns)
+		framework.ExpectNoError(err)
+
+		ns = updatedNS.Name
+
+		err = clearDaemonSetNodeLabels(c)
+		framework.ExpectNoError(err)
+	})
+
+	ginkgo.It("should manage the lifecycle of a ControllerRevision", func() {
+		csAppsV1 := f.ClientSet.AppsV1()
+
+		dsLabel := map[string]string{"daemonset-name": dsName}
+		dsLabelSelector := labels.SelectorFromSet(dsLabel).String()
+
+		ginkgo.By(fmt.Sprintf("Creating DaemonSet %q", dsName))
+		testDaemonset, err := csAppsV1.DaemonSets(ns).Create(context.TODO(), newDaemonSetWithLabel(dsName, image, dsLabel), metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Check that daemon pods launch on every node of the cluster.")
+		err = wait.PollImmediate(dsRetryPeriod, dsRetryTimeout, checkRunningOnAllNodes(f, testDaemonset))
+		framework.ExpectNoError(err, "error waiting for daemon pod to start")
+		err = e2edaemonset.CheckDaemonStatus(f, dsName)
+		framework.ExpectNoError(err)
+
+		ginkgo.By(fmt.Sprintf("Confirm DaemonSet %q successfully created with %q label", dsName, dsLabelSelector))
+		dsList, err := csAppsV1.DaemonSets("").List(context.TODO(), metav1.ListOptions{LabelSelector: dsLabelSelector})
+		framework.ExpectNoError(err, "failed to list Daemon Sets")
+		framework.ExpectEqual(len(dsList.Items), 1, "filtered list wasn't found")
+
+		ds, err := c.AppsV1().DaemonSets(ns).Get(context.TODO(), dsName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By(fmt.Sprintf("Listing all ControllerRevisions with label %q", dsLabelSelector))
+		revs, err := csAppsV1.ControllerRevisions("").List(context.TODO(), metav1.ListOptions{LabelSelector: dsLabelSelector})
+		framework.ExpectNoError(err, "Failed to list ControllerRevision: %v", err)
+		framework.ExpectNotEqual(len(revs.Items), 0, "Failed to find any controllerRevisions")
+
```

Comment:
do we expect more than 1 revision?

Reply:
No, it was a leftover from some earlier testing. Fixed.

Reply:
in that case we can also remove the loop

Reply:
Fixed.